<launch>
  <!--Publish a pcd as pointcloud and run clustering node, must provide tf-->
  <arg name="name" default="takeshiBagPack1"/>
  <arg name="crop" default="2.0"/>
  <arg name="freeSpace" default="0.1"/>


  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
    <param name="resolution" value="0.03"/>

    <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
    <param name="frame_id" type="string" value="base_link"/>
    <remap from="octomap_point_cloud_centers" to="/hsrb/voxelmap_cloud" />
    <!-- maximum range to integrate (speedup!) -->
    <param name="sensor_model/max_range" value="5.0"/>
    <param name="filter_ground " value="false" type="bool"/>
    <param name="base_frame_id " value="base_link" type="str"/>

    <!-- data source to integrate (PointCloud2) -->
    <remap from="cloud_in" to="/hsrb/cloud"/>

  </node>

  <node name="space_sep_vox" pkg="space_quantization" type="space_separation_voxel_node" output="screen">
    <remap from="voxelmap_cloud" to="/hsrb/voxelmap_cloud"/>
    <remap from="free_voxels" to="/hsrb/free_voxels"/>
    <remap from="occupied_voxels" to="/hsrb/occupied_voxels"/>
    <param name="free_thr" value="$(arg freeSpace)" type="double"/>
    <param name="pub_seg_space" value="true" type="bool"/>
  </node>

  <node name="adjGraph" pkg="space_quantization" type="adjacency_graph_node" output="screen">
    <remap from="voxelized_space" to="/hsrb/free_voxels"/>
    <remap from="graph_marker" to="/hsrb/graph_marker"/>
    <param name="k_neighboors" value="6" type="int"/>
    <param name="max_dist" value="0.65" type="double"/>
    <param name="graph_file" value="$(find space_quantization)/graphs/$(arg name)_voxel.txt" type="str"/>
  </node>

  <!-- <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_table" args="-0.005 1.115 0.802 -0.037 0.821 -0.569 -0.015 base_link head_rgbd_sensor_rgb_frame 100"/> -->

  <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_backpack" args="0.016 -0.073 0.979 -0.009, 0.887, -0.462, -0.004 base_link head_rgbd_sensor_rgb_frame 100"/>

  <!-- <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_coyotzi" args="0.012, 0.022, 1.323 -0.085, 0.772, -0.627, 0.057 base_link head_rgbd_sensor_rgb_frame 100"/> -->

  <!-- <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_byron" args="0.020, -0.010, 1.326, -0.139, 0.810, -0.563, 0.086 base_link head_rgbd_sensor_rgb_frame 100"/> -->
  <!-- head_rgbd_sensor_rgb_frame -->
  <node name="cloudPublisher" pkg="pcl_ros" type="pcd_to_pointcloud" args="$(find space_quantization)/cloud/$(arg name).pcd " output="screen">
    <remap from="cloud_pcd" to="/hsrb/cloud"/>
    <param name="frame_id" value="head_rgbd_sensor_rgb_frame" type="str"/>
  </node>

  <node name="viewr" pkg="pcl_ros" type="convert_pointcloud_to_image" args="" output="screen">
    <remap from="input" to="/hsrb/cloud"/>
    <remap from="output" to="/hsrb/head_rgbd_sensor/rgb/image_raw"/>

  </node>

</launch>
